File: src/tests/Markify.Fixtures/Projects/Source/Class/ClassMethods.vb
Public Class FooType
    Sub Method()
	End Sub

    Public Sub PublicMethod()
	End Sub

    Friend Protected Sub InternalProtectedMethod()
	End Sub

    Function IntMethod() As Integer
		Return 1
	End Function

    Public Overridable Sub VirtualMethod()
	End Sub

    Private Partial Sub PartialMethod()
    End Sub

    Sub WithParametersMethod(foo As Integer, ByRef bar As Integer, ByVal foobar As Integer)
	End Sub

    Function SingleGenericMethod(Of T)(Optional foo As T = Nothing) As T
		Return CType(Nothing, T) 
	End Function

    Function MultiGenericMethod(Of T As IList, Y As {IDisposable, IEnumerable})(foo As T, bar As y) As T
		Return CType(Nothing, T) 
	End Function
End Class

File: src/Services/Roslyn/Markify.Services.Roslyn.Fixtures/Projects/Source/Class/ClassMethods.cs
public partial class FooType
{
    void Method() { }

    public void PublicMethod() { }

    internal protected void InternalProtectedMethod() { }

    int IntMethod() { return 1; }

    public virtual void VirtualMethod() { }

    partial void PartialMethod();

    void WithParametersMethod(int foo, ref int bar, out int foobar) { foobar = 2; }

    void WithNoNameParameterMethod(__arglist) { }

    T SingleGenericMethod<T>(T foo = default(T)) { return default(T); }

    T MultiGenericMethod<T, Y>(T foo, Y bar) 
        where T : IList 
        where Y : IDisposable, IEnumerable
    {
        return default(T);
    }

    int BodyMethod(int foo) => 1;
}